python/pyspark/sql/tests/connect/test_connect_basic.py

```diff
@@ -18,6 +18,9 @@
 import unittest
 import shutil
 import tempfile
+
+import grpc  # type: ignore
```

Comment:
Maybe we can do this separately but we should have a logic like `have_pandas` for the main codes too.
The main problem is that PySpark itself doesn't have anything as a required dependency, and difficult to add a hard required dependency. e.g., you download Spark binary from https://spark.apache.org/downloads.html in the cluster, and the basic functionality should work. If `grpc` is not installed, it should show the proper exception message.

Reply:
I think this comes back to the discussion of how we want to isolate the client from the rest. In PySpark proper it will simply fail with an import error when you call `df.toPandas()` because the import is only done there. 

Let me see if I can move the imoprt arround a bit.

Reply:
I moved the pandas import arround :) but the grpc import cannot be really moved because even the proto generated code has a grpc dependency and will fail there.